coding-intro/coding-intro-core/core-concepts/programs-and-data.md

```diff
@@ -8,35 +8,77 @@ category: must-know
 links:
   - >-
     [Exact Instructions Challenge](https://youtube.com/watch?v=Ct-lOOUqmyY){video}
-  
+practiceQuestion:
+  formats:
+    - fill-in-the-gap
+  context: standalone
+
 ---
 
 # Programs & Data
 
 ---
 ## Content
 
-Let's quickly introduce the building blocks we'll cover in this course.
+Let's introduce the building blocks we'll cover in this course.
+
+
+### Programs
+
+Today you might have woken up with a to-do list, which is essentially your program for the day.
+
+Similarly, we give the computers a to-do list of extremely specific commands, together called a program, to carry out.
+
+
+> 💡 Scroll down to the Learn more section to see a hilarious example of how specific instructions computers need 😆
+
+
+### Data
 
-### Program
+Data is the information that our programs work with. In code, we store these in variables.
 
-Today you woke up with a plan, a to-do list, which is essentially your program for the day. Similarly, we give the computers a set of commands to carry out, one after the other. That too is known as a program.
+For instance, we could store someone's `name` and `age` in variables[1]:
+```python
+name = "Batman"
+age = 30
+```
 
-The only difference being that computers require specific instructions. (Don't tell anyone we said this, but machines are actually pretty dumb… sssh!)
 
-> 💡 Scroll down to see a hilarious real life example of this in the Learn more section 😆
+### Types
 
+Now remember, computers are extremely literal machines! If you write a program to divide "Batman" by 2, it is going to try (and fail) to do that.
```

Comment:
This sentence is good but not sure it's an improvement.

> It’s easy to divide 24 by 2 and get 12. However, dividing “Batman” by 2 might land you in jail.

I'd consider leaving this playful copy in there somehow. IIRC, it generated some fun engagement.

Lets dig into the comments on this insight and see what people say?